### PR TITLE
Modified firefile constructor to meet China and Global file upload

### PR DESCRIPTION
**Pull request by techuanliu**

### CODE REVIEW

File: src/main/java/com/invisibi/firefile/FireFileController.java [hunk 1]
```diff
@@ -106,21 +109,27 @@ public void clearCache() {
...
+                ? userFolder + File.separator + UUID.randomUUID() + "-" + state.name()
+                : UUID.randomUUID() + "-" + state.name();
         if (!TextUtils.isEmpty(state.mimeType())) {
             objectId = objectId.replace(".tmp", "." + state.mimeType());
         }
         final String finalObjectId = objectId;
-        TransferObserver transferObserver = transferUtility.upload(s3Bucket, DEFAULT_SUB_FOLDER + File.separator + objectId, file, CannedAccessControlList.PublicReadWrite);
+        TransferObserver transferObserver = transferUtility
+                .upload(s3Bucket, DEFAULT_SUB_FOLDER + File.separator + objectId, file,
+                        CannedAccessControlList.PublicReadWrite);
```
Comment (tivonahu) [inline src/main/java/com/invisibi/firefile/FireFileController.java]: ACL should be PublicRead

File: src/main/java/com/invisibi/firefile/FireFile.java [hunk 2]
```diff
@@ -124,18 +124,26 @@ public String url() {
...
+    }
+
+    public static void initialize(final Context context, final String awsIdentityPoolId, final String s3URL,
+                                  final String s3Bucket, final Regions s3Regions, final boolean useUserFolder) {
         final CognitoCachingCredentialsProvider
                 credentialsProvider = new CognitoCachingCredentialsProvider(context, awsIdentityPoolId, s3Regions);
         final AmazonS3Client s3Client = new AmazonS3Client(credentialsProvider);
-        initialize(context, s3Client, s3URL, s3Bucket);
+        final String userFolder = useUserFolder
+                ? credentialsProvider.getIdentityId()
```
Comment (tivonahu) [inline src/main/java/com/invisibi/firefile/FireFile.java]: Note: Should avoid calling getIdentityId() in main thread
http://docs.aws.amazon.com/cognito/latest/developerguide/getting-credentials.html

File: src/main/java/com/invisibi/firefile/FireFile.java [hunk 3]
```diff
@@ -124,18 +124,26 @@ public String url() {
     }
 
     public static void initialize(final Context context, final AmazonS3Client amazonS3Client,
-                                  final String s3URL, final String s3Bucket) {
-        FireFile.fFileController = new FireFileController(context, amazonS3Client, s3URL, s3Bucket);
+                                  final String s3URL, final String s3Bucket, final String userFolder) {
+        FireFile.fFileController = new FireFileController(context, amazonS3Client, s3URL, s3Bucket, userFolder);
```
Comment (tivonahu) [inline src/main/java/com/invisibi/firefile/FireFile.java]: If this is for China, must be sure you setupped AWS auth (AmazonS3Client) before initializing this. Or move auth control here.

File: src/main/java/com/invisibi/firefile/FireFileController.java (hunk 1 above)
Comment (techuanliu) [inline src/main/java/com/invisibi/firefile/FireFileController.java]: Fixed

File: src/main/java/com/invisibi/firefile/FireFile.java (hunk 2 above)
Comment (techuanliu) [inline src/main/java/com/invisibi/firefile/FireFile.java]: code refined, credentialsProvider.getIdentityId() is no longer use in FireFile, credentialsProvider.getIdentityId() is call in background thread and value is passed from app.

File: src/main/java/com/invisibi/firefile/FireFile.java (hunk 3 above)
Comment (techuanliu) [inline src/main/java/com/invisibi/firefile/FireFile.java]: yes, the logic is done in Sunray app actually

**Pull request merged**